### ISSUE TITLE
C# client throwing exception trying to Deserialize a json response into a string

### ISSUE THREAD

**Issue by jlewis-rhimat** [RSuter/NSwag#1104]
I am using an externally generated swagger.json definition that returns json data as a string. The client is failing with an exception calling DeserializeObject in the following code
` if (status_ == "200")
                        {
                            var responseData_ = await response_.Content.ReadAsStringAsync().ConfigureAwait(false);
                            var result_ = default(string);
                            try
                            {
                                result_ = Newtonsoft.Json.JsonConvert.DeserializeObject<string>(responseData_, _settings.Value);
                                return result_;
                            }
                            catch (System.Exception exception_)
                            {
                                throw new SwaggerException("Could not deserialize the response body.", status_, responseData_, headers_, exception_);
                            }
                        }`
The call to DeserializeObject fails (and is not required) when the result is also a string. The result should just be set to the reponseData value. 

Whats the best way to accomplish this? Do I need to update templates?

Thanks.

**Comment by RSuter** [comment]
The spec is most probably wrong - I assume that it responds with "application/json" and thus a string like `foobar` is not allowed (it is not valid JSON) but must have "... If you return plain text you have to change the spec to "text/plain" and this should generate code without JSON deserializiation...

**Comment by jlewis-rhimat** [comment]
The returned data is valid json and can be parsed using JObject.Parse ok.

**Comment by RSuter** [comment]
A string like `foobar` is not valid JSON, only `"foobar"` is

**Comment by jlewis-rhimat** [comment]
The returned string is properly formatted json
   ` {"dx_deltaJSON":{"dx_data_sets" ... }}
`
I've checked it can be parsed using JObject.Parse and verified its ok on the Json.Lint website.

The call to `result_ = Newtonsoft.Json.JsonConvert.DeserializeObject<string>(responseData_, _settings.Value);` will fail when the result type is a string.

**Comment by RSuter** [comment]
Of course, `{"dx_deltaJSON":{"dx_data_sets" ... }}` is not a string but a complex object. The response however expects a JSON encoded string in the form `"asdf"`... Can you post the swagger spec of the operation?

**Comment by jlewis-rhimat** [comment]
I'll look into posting the definition. The response is a valid Json encoded string, see snippet from debug session.
![image](https://user-images.githubusercontent.com/13539211/34168640-f0c390a6-e4dc-11e7-89e4-c61a6e49b992.png)

**Comment by RSuter** [comment]
Yes, but the Swagger spec describes the response as a single string JSON token, not a complex type with properties like dx_deltaJSON, etc

**Comment by jlewis-rhimat** [comment]
The swagger definition is available at

https://bitbucket.org/deltaxml/json-compare/raw/681462e9e1ee4fc5661444081e36ebbd9194c6e6/swagger.json

I'm using the "/compare/multi_part_strings" path that has a 200 response defined as
 ```
"responses": {
          "200": {
            "description": "successful operation",
            "schema": {
              "type": "string"
            }

```          },

In this case the response is a delta json representation of the differences between two json inputs that can have any valid json format. The api just need to pass the string result through to the consumer without further deserialization.

**Comment by jlewis-rhimat** [comment]
I'm not an expert but the swagger 2.0 spec says the response structure (as defined in the schema) can be a primitive, object or array. In this case the response schema is type string (primitive). 
The current implementation using 
     `string result_ = Newtonsoft.Json.JsonConvert.DeserializeObject<string>()`
 will always throw an exception in this circumstance. 

My suggested fix is to tweak the C#Client template Client.Class.ProcessResponse.liquid to handle this scenario. See 
![template_fix](https://user-images.githubusercontent.com/13539211/34522394-4d9f7700-f08a-11e7-8089-ee5f440abb95.PNG)

**Comment by jlewis-rhimat** [comment]
Here is the updated template

[Client.Class.ProcessResponse.liquid.zip](https://github.com/RSuter/NSwag/files/1600524/Client.Class.ProcessResponse.liquid.zip)

**Comment by ischyrus** [comment]
I just started to see this issue too. I think it is fairly new.

**Comment by jlewis-rhimat** [comment]
Any more thoughts on this issue and suggested fix?

**Comment by ischyrus** [comment]
I wrapped my response in an object called "[MyRequest]Response". Probably a better pattern anyway.

**Comment by berkerdong** [comment]
```
public void ConfigureServices(IServiceCollection services)
{
    services.AddControllers(options =>
    {
        // requires using Microsoft.AspNetCore.Mvc.Formatters;
        options.OutputFormatters.RemoveType<StringOutputFormatter>();
        options.OutputFormatters.RemoveType<HttpNoContentOutputFormatter>();
    });
}
```

![image](https://user-images.githubusercontent.com/36178221/142809051-167070a4-ce56-4201-a575-65f3c8acef9d.png)

[https://docs.microsoft.com/en-us/aspnet/core/web-api/advanced/formatting?view=aspnetcore-6.0](url)

This can be fixed on the server side of the API

**Comment by willhuo** [comment]
I found a solution in stack overflow ,https://stackoverflow.com/questions/67254632/client-from-nswag-csharpclientgenerator-cannot-deserialize-string

The issue is that your returning string is not in quotes, that is the reason why you're getting deserialize error. To fix it just remove .NET :HttpNoContentOutputFormatter

services.AddControllers(options =>
    {
        options.OutputFormatters.RemoveType<StringOutputFormatter>();
    });
As a result the service will produce content (with quotes), which is valid string type now."Pong3"
